sympy/functions/elementary/piecewise.py

```diff
@@ -512,7 +512,10 @@ def handler(ipw):
                 cond = (x < b)
             else:
                 cond = (x <= b)
-            args.append((sum, cond))
+            if sum.has(oo, -oo, S.ComplexInfinity, Undefined):
```

Comment:
How about testing infinities on line 504?

Reply:
Testing infinities on line 504 does not catch infinities produced on line 506. I could change it so that it tests infinities right line 506 if that would help somehow.

Reply:
I'd suggest using `S.NegativeInfinity` instead of `-oo` to speed up the code (although both are very fast, an expression is created for the current code). An therefore also `S.Infinity` for consitency. What about `S.NaN`?

Reply:
If `sum` is infinite or undefined, then it should probably be reset to 0.

Reply:
> If `sum` is infinite or undefined, then it should probably be reset to 0.

Yes, this seems like a good idea. I'll add it in along with oscargus's suggestion.

Reply:
> I'd suggest using `S.NegativeInfinity` instead of `-oo` to speed up the code (although both are very fast, an expression is created for the current code). An therefore also `S.Infinity` for consistency. What about `S.NaN`?


I'll replace `oo` and `oo` with `S.NegativeInfinity` and `S.Infinity` respectively. In this file, `Undefined` is set to `S.NaN`, so `S.NaN` is already covered.